drivers/storage/portworx/component/security.go

```diff
@@ -0,0 +1,212 @@
+package component
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+
+	"github.com/hashicorp/go-version"
+	pxutil "github.com/libopenstorage/operator/drivers/storage/portworx/util"
+	corev1alpha1 "github.com/libopenstorage/operator/pkg/apis/core/v1alpha1"
+	k8sutil "github.com/libopenstorage/operator/pkg/util/k8s"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// SecurityComponentName is the name for registering this component
+	SecurityComponentName = "Security"
+	// SecurityPXPrivateKeysSecretName is the admin secret name for PX security
+	SecurityPXPrivateKeysSecretName = "px-private-keys"
```

Comment:
Can we call this something different? Not sure what but does not convey that this has secrets for auth. We can talk about this offline.